File: backend/Testing/Browser/Page/RegisterPage.cs
using Microsoft.Playwright;

namespace Testing.Browser.Page;

public class RegisterPage : BasePage<RegisterPage>
{
    public RegisterPage(IPage page)
    : base(page, "/register", page.GetByRole(AriaRole.Heading, new() { Name = "Register" }))
    {
    }

    public async Task FillForm(string name, string email, string password)
    {
        await Page.GetByLabel("Name").FillAsync(name);
        await Page.GetByLabel("Email").FillAsync(email);
        await Page.GetByLabel("Password").FillAsync(password);
    }

    public async Task Submit()
    {
        await Page.GetByRole(AriaRole.Button, new() { Name = "Register" }).ClickAsync();
    }
}

File: frontend/tests/pages/registerPage.ts
import type { Page } from '@playwright/test';
import { BasePage } from './basePage';

export class RegisterPage extends BasePage {
  constructor(page: Page) {
    super(page, page.getByRole('heading', {name: 'Register'}), `/register`);
  }

  async fillForm(name: string, email: string, password: string): Promise<void> {
    await this.page.getByLabel('Name').fill(name);
    await this.page.getByLabel('Email').fill(email);
    await this.page.getByLabel('Password').fill(password);
  }

  async submit(): Promise<void> {
    await this.page.getByRole('button', {name: 'Register'}).click();
  }
}
